challenges/backend/app/data.py

```diff
@@ -0,0 +1,163 @@
+		"""
+		# does the post have a json param:
+		if new_request.is_json is False:
+			raise exceptions.BadRequest("Data must be a JSON formatted object.")
+		# get the JSON from the request
+		new_data, is_dict = cls.validate_type(new_request)
+
+		if is_dict:
+			for key, numbers in new_data.items():
+				new_data[key] = cls.validate_numbers(numbers)
+			cls.values.data = new_data
+		else:
+			cls.values.data = cls.validate_numbers(new_data)
+
+		# return to flask response object, type json
+		return jsonify(cls.values.data), 201
+
+	@classmethod
+	def validate_type(cls, new_request):
+		# does the post have a json param:
+		if new_request.is_json is False:
+			raise exceptions.BadRequest("Data must be a JSON formatted object.")
+		# get the JSON from the request
+		new_data = new_request.get_json()
+
+		# check type: dict
+		if isinstance(new_data, dict):
+			if len(new_data.keys()) == 1:
+				is_dict = True
+			else:
+				raise exceptions.BadRequest(
+					"JSON is in incorrect format. Object must have either 0 or 1 keys and a 500-length array.")
+		# check type: list
+		elif isinstance(new_data, list):
+			is_dict = False
+		# raise for unaccounted-for formatting
+		else:
+			raise exceptions.BadRequest(
+				"Something is wrong with the data."
+				"Please reformat the data as a JSON object with 0 or 1 keys, and a 500-length array of numbers.")
+
+		return new_data, is_dict
+
+	@staticmethod
+	def validate_numbers(numbers):
+		"""
+		checks if the array sent contains 500 numbers amd raises an error if the data is invalid
+		:param numbers: a list of data that should be numbers
+		:returns: if there are no errors, returns a list of numbers sorted by value
+		"""
+		if len(numbers) != 500:
+			raise exceptions.BadRequest("Wrong length.")
+		# validate: list content
+		else:
+			flag_good = True
+			for i in range(len(numbers)):
+				if not isinstance(numbers[i], (int, float)):
+					flag_good = False
+					raise exceptions.BadRequest("Values must be numbers.")
+			# set data if valid
+			# todo: deal with data_key
+			if flag_good is True:
+				return sorted(numbers)
+
+	@classmethod
+	def patch_data(cls, patch: request):
+		"""
+		patches the array in values.data with a random number,
+		and removes the largest number from the list.
+		there is likely a cleaner way to make this function,
+		but this prevents assignment of data via values.seq.
+		:param patch:
+		:return: a json object of the new data and response code 204
+		"""
+		# raises an error if there's no data present
+		if not cls.values.data or cls.values.data == [[]]:
+			raise exceptions.NotFound("There's no data here yet!")
+		# gets the array, removes the highest value, and appends a new random int
+		else:
+			new_data = cls.values.seq
+			del new_data[499]
+			new_data.append(random.randint(-1000, 1000))
+			# reassigns the data if dict
+			if cls.values.is_dict:
+				for key, val in cls.values.data.items():

```

Comment:
What happens when a request comes in to your server to `set_data` while this line(149) is executing?  Assume the client actually wants to `set_data` to a new sequence while line 149 is still executing on the original list while `set_dat`a is called with a new sequence.
How do you solve this problem? (No need to write any code, just an explanation)